st2common/st2common/models/api/action.py

```diff
@@ -682,3 +683,51 @@ def to_model(cls, aliasexecution):
     @classmethod
     def from_model(cls, aliasexecution):
         raise NotImplementedError()
+
+
+class ActionAliasHelpAPI(BaseAPI):
+    """
+    API model used to display action-alias help API endpoint.
+    """
+    model = None
+
+    schema = {
+        "title": "ActionAliasHelpAPI",
+        "description": "ActionAliasHelpAPI.",
+        "type": "object",
+        "properties": {
+            "filtering": {
+                "type": "string",
+                "description": "Find help strings containing keywords.",
```

Comment:
The intention was to have the string represent a regular expression pattern which provides very powerful search features.

You can get a similar effect as a list by using regex pipes. e.g. `item1|item2|item3`.  There is a limitation however, the API doesn't allow for passing regex flags with the request, so using case insensitive search isn't currently possible.

That said, I'll use whatever search mechanism is recommended by the st2 devs.